### PR TITLE
Added week4.tex

### PR DESCRIPTION
**Pull request by ktbooker**
This is our submission for the week 4 notes. Please let me know if anything needs to be fixed.

### CODE REVIEW

**Review by HarrisonGrodin — APPROVED** [review]
Looks great! Only issue seems to be that `intuition.png` is missing?

File: week4.tex [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+Note that $x$ refers to \textit{term variables}, whereas $X$ refers to \textit{type variables}.
+
+System F mostly copies the statics and dynamics from the STLC, with the addition of some extra rules for incorporating type variables. Those are replicated below:
+
+Static rules relating to quantifiers:
+
+$$
+\defrule[Lam][sta:lam]
+  {\Gamma \entails[\Delta, X]{\isOfTp{M}{A}}}
+  {\Gamma \entails[\Delta]{\isOfTp{\tlamEx*{X}{M}}{\allTy*{X}{M}}}}
```
Comment (EULIR) [inline week4.tex:32]: I believe `\Lambda` needs to be final. See PFPL 16.3b.

**Review by EULIR — CHANGES_REQUESTED** [review]

File: week4.tex [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  {\Gamma \entails[\Delta]{\isOfTp{M}{\allTy*{X}{A}}} \\
+   \Delta \entails{B \text{ type}}}
+  {\Gamma \entails[\Delta]{\isOfTp{\tapEx*{M}{B}}{\Sub{B}{X}{A}}}}
+$$
+
+Dynamic rules relating to quantifiers:
+$$
+\defrule[Lam][dyn:lam]
+  {M \stepsTo M'}
+  {\tlamEx*{X}{M} \stepsTo{} \tlamEx*{X}{M'}}
```
Comment (runmingl) [inline week4.tex]: I believe `\Lambda` needs to be final. See PFPL 16.3b.

**Review by runmingl — CHANGES_REQUESTED** [review]

File: week4.tex (hunk 2 above)
Comment (HarrisonGrodin) [reply week4.tex]: Oh, very true, nice catch!

**Review by runmingl — APPROVED** [review]

**Pull request merged**